### ISSUE TITLE
publish on AltStore PAL for iOS/iPads

### ISSUE THREAD

**Issue by RootLUG** [vcmi/vcmi#4434]
Hello,
I tried to sideload the ios version and noticed AltStore PAL allows adding external sources which can be used to install and auto-update this app without using the problematic self-signing with expiration in 7 days.
I would like to discuss if it would be possible to publish this app on AltStore so EU users can install it more easily.

I understand it still requires a developer profile on Apple to get it signed which involves fees. Hypothetically I already have it as I own a small consultancy company where I bought it to experiment with porting some software to ios and I would be willing to cover it on that side as I can re-use my dev profile to submit it for signing so it can be published.

What else would be needed to be able to publish it on AltStore? From the documentation it appears it can then be hosted on webserver once properly signed. Are there any other roadblocks involved than the fees associated with getting the app signed?

**Comment by kambala-decapitator** [comment]
well, you can already use "classic AltStore" to solve the 7 days signing for you.

no offense, but using dev account of an outsider doesn't sound like a great idea. But since you already have dev acocunt, you don't have to suffer from the 7 days limit - simply sign VCMI with your paid certificate and a wildcard provisioning profile, then you'll need to resign only once a year.

> Are there any other roadblocks involved than the fees associated with getting the app signed?

the only thing preventing us from publishing on AppStore (or AltStore) is the Apple developer account yearly fee.

**Comment by RootLUG** [comment]
I understand. One more thought, if I would pay the dev profile fees on behalf of someone else so it can be published by a developer account officially affiliated with the project; would it be possible in exchange to place something along the lines that "publishing this version was sponsored by <company_name>" ? Somewhere in downloads/ios section or description on the store.

I don't mean it in a way to push any game ads, I absolutely hate those. I just have the possibility of expensing those on behalf of the company if I provide a valid reason for accounting in case of audit when they will ask "what was the purpose of this".

If you have any other ideas how this can be arranged so it has "proper accounting" then please let me know so more people can enjoy this fantastic game on the go.

**Comment by kambala-decapitator** [comment]
@IvanSavenko @DjWarmonger @dydzio0614 @ArseniyShestakov please check the above comment

**Comment by dydzio0614** [comment]
OK for me

**Comment by IvanSavenko** [comment]
Not against it, but personally I would rather set up donations, as discussed before: https://github.com/vcmi/vcmi/issues/3795
and use them to fund apple license.

Was planning to do so soon, since apparently nobody else is interested in setting it up and I have some free-ish time.

@kambala-decapitator what do we need to do in total? Is Apple device (mac/iphone) needed for any of these steps?
- Create Apple account (for "VCMI Team" I guess?)
- Pay for developer license
- Modify our CI to sign ipa's / dmg's
- Publish on AltStore
- Publish on Apple Store?

**Comment by kambala-decapitator** [comment]
> * Create Apple account (for "VCMI Team" I guess?)

yes, VCMI team would be perfect. But idk the amount of effort to create account for an organisation compared to an individual.

> Is Apple device (mac/iphone) needed for any of these steps?

macOS is needed to upload builds to AppStore (iOS / macOS) and to notarize macOS builds (when distributing outside of Mac App Store). Signing can technically be performed on any OS, but on macOS it's easier. Publishing itself (pressing a button) is done in a browser.

**Comment by IvanSavenko** [comment]
> But idk the amount of effort to create account for an organisation compared to an individual.

Yeah, seems to require legal entity. And individual account seems to require 2FA so not something I'll be able to create in "VCMI team" name.

> macOS is needed to upload builds to AppStore (iOS / macOS) and to notarize macOS builds 

Can we use Github macOS CI's for that?

**Comment by kambala-decapitator** [comment]
> Can we use Github macOS CI's for that?

Sure
